File: frontend/src/types/auditLog.ts
export interface AuditLogItem {
  id: number
  actor_id: number | null
  actor_username: string | null
  actor_full_name: string | null
  actor_role: number | null
  action: string
  target_type: string | null
  target_id: string | null
  target_name: string | null
  ip_address: string | null
  is_success: boolean
  details: string | null
  created_at: string
}

export interface AuditLogListResponse {
  items: AuditLogItem[]
  total: number
  page: number
  page_size: number
}

export interface AuditLogQueryParams {
  page?: number
  page_size?: number
  action?: string
  actor_id?: number
  target_type?: string
  is_success?: boolean
  from_date?: string
  to_date?: string
  search?: string
}

export const AUDIT_ACTION_LABELS: Record<string, { label: string; color: string }> = {
  LOGIN_SUCCESS: { label: 'Đăng nhập thành công', color: 'green' },
  LOGIN_FAILED: { label: 'Đăng nhập thất bại', color: 'red' },
  USER_ACTIVATE: { label: 'Kích hoạt tài khoản', color: 'blue' },
  USER_DEACTIVATE: { label: 'Khóa tài khoản', color: 'orange' },
  ROLE_CHANGE: { label: 'Đổi vai trò', color: 'purple' },
  CLEARANCE_CHANGE: { label: 'Cấp/Thu hồi cơ mật', color: 'amber' },
  DIRECTIVE_CHANNEL_ACCESS: { label: 'Quyền Kênh chỉ đạo', color: 'teal' },
  COMMAND_CHANNEL_ACCESS: { label: 'Quyền Kênh chỉ huy', color: 'indigo' },
  PASSWORD_RESET: { label: 'Đặt lại mật khẩu', color: 'pink' },
  DISPATCH_DOWNLOAD: { label: 'Tải công văn mật', color: 'red' },
}

File: backend/app/schemas/audit_log.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict


class AuditLogCreate(BaseModel):
    actor_id: Optional[int] = None
    actor_username: Optional[str] = None
    actor_full_name: Optional[str] = None
    actor_role: Optional[int] = None
    action: str
    target_type: Optional[str] = None
    target_id: Optional[str] = None
    target_name: Optional[str] = None
    ip_address: Optional[str] = None
    is_success: bool = True
    details: Optional[str] = None


class AuditLogOut(BaseModel):
    id: int
    created_at: datetime
    actor_id: Optional[int] = None
    actor_username: Optional[str] = None
    actor_full_name: Optional[str] = None
    actor_role: Optional[int] = None
    action: str
    target_type: Optional[str] = None
    target_id: Optional[str] = None
    target_name: Optional[str] = None
    ip_address: Optional[str] = None
    is_success: bool
    details: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class AuditLogListResponse(BaseModel):
    items: list[AuditLogOut]
    total: int
    page: int
    page_size: int
